components/dashboard/src/service/service.ts

```diff
@@ -52,4 +42,38 @@ function createGitpodService<C extends GitpodClient, S extends GitpodServer>() {
     return service;
 }
 
-export const ServiceContext = React.createContext<Service>(undefined! as Service /* we need to pass in the value on the root when using <ServiceContext.Provider />. See index.tsx */);
\ No newline at end of file
+
+
+export class AppService {
+    constructor(protected gitpodService: GitpodService) {
+    }
+
+    protected userPromise: Promise<User> | undefined;
+    async getOrLoadUser() {
+        if (!this.userPromise) {
+            this.userPromise = this.gitpodService.server.getLoggedInUser();
+        }
+        return this.userPromise;
+    }
+    async reloadUser() {
+        this.userPromise = undefined;
+        return this.getOrLoadUser();
+    }
+
+    async getAuthProviders() {
+        return this.gitpodService.server.getAuthProviders();
+    }
+
+}
+
+let gitpodService: GitpodService;
+let service: AppService;
```

Comment:
how about storing the userPromise here instead, we might be able to go without an `AppService`.